ui/app/appointments/controllers/manage/calendar/appointmentsCalendarViewController.js

```diff
@@ -117,14 +117,15 @@ angular.module('bahmni.appointments')
                     }
                 }
             }, true);
-
             var setAppointments = function (params) {
                 autoRefreshStatus = false;
                 return appointmentsService.getAllAppointments(params).then(function (response) {
                     $scope.allAppointmentsForDay = response.data;
                     var filteredAppointments = appointmentsFilter($scope.allAppointmentsForDay, $state.params.filterParams);
                     $rootScope.appointmentsData = filteredAppointments;
                     autoRefreshStatus = true;
+                    $rootScope.appointmentStartTime = _.last($scope.allAppointmentsForDay);
```

Comment:
@prashantsable1991 Here it should find the earliest slot not the one available after the last appointment of the day